CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.10)
```

Comment:
> It is breaking compatibility with e.g. CentOS 7 (maintained up to 2024 and still a lot used on servers), which has only CMake 2.8.11 (current minimum version).

Is CentOS being used on DV capture machines?  I'd understand server support if this was a library which other tools are built on but this is an end-user tool, right? 

If you NEED to compile on a CentOS 7 for CentOS, you can easily download a portable binary for Linux directly from their website if you really want to support old systems. 

If you're only wanting to run a compiled binary, such as for package managers, the end users doesn't need CMake to install and run it. Just build a binary package on a machine with a modern toolchain, and install that package on the machine that doesn't. 

There are many reasons to not use legacy CMake. it is horrible in so many ways. It's so easy to introduce a bug that fails silently because it relies on flakey variable expansion and setting compiler flags directly. Modern CMake (see this presentation from BoostCon https://www.youtube.com/watch?v=bsXLMQ6WgIk) is much better because targets (executables and libraries) have properties which CMake can generate the right compiler flags for the toolchain during config or build time. Legacy CMake did it somewhat, but it wasn't very robust. Modern CMake is significantly more maintainable and scale very well.

Honestly, since this is a new project, I would love to use the most recent CMake standard since it's very powerful (the version being released any day now supports PCH), and it makes generating .deb, .msi, .rpm, and .dmg packages for distribution much easier.

Reply:
This is a new project, but there's an intent to follow precedent with our mediaarea projects in regards to how they are built. In considering the project's sustainability and ongoing maintenance I think I'd prefer to maintain the similarities.